### PR TITLE
merge Parallelize calculate root hash into RC3

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description

This PR essentially makes the `release-22.10.x` branch the same as `main` with #4568 merged into it

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java [hunk 1]
```diff
@@ -219,31 +218,37 @@ DataFetcher<Optional<AccountAdapter>> getAccountDataFetcher() {
...
+            .or(
+                () -> {
+                  if (bn > blockchainQuery.getBlockchain().getChainHeadBlockNumber()) {
+                    // block is past chainhead
+                    throw new GraphQLException(GraphQLError.INVALID_PARAMS);
+                  } else {
+                    // we don't have that block
+                    throw new GraphQLException(GraphQLError.CHAIN_HEAD_WORLD_STATE_NOT_AVAILABLE);
+                  }
+                });
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:240]: I wasn't expecting GraphQL changes in this merge

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:240]: oh it's the getAndMapWorldState change

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java [hunk 2]
```diff
@@ -49,7 +49,6 @@ public enum RpcMethod {
   ENGINE_NEW_PAYLOAD("engine_newPayloadV1"),
   ENGINE_FORKCHOICE_UPDATED("engine_forkchoiceUpdatedV1"),
   ENGINE_EXCHANGE_TRANSITION_CONFIGURATION("engine_exchangeTransitionConfigurationV1"),
-  ENGINE_PREPARE_PAYLOAD("engine_preparePayload_debug"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:52]: ?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java (hunk 1 above)
Comment (garyschulte) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:240]: yeah, those are from the snapshot bonsai PR that merged to main.  The version of the snapshot pr that was originally in rc3 did not have those changes

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java (hunk 2 above)
Comment (garyschulte) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:52]: the RC3 branch originally had #4427 merged in to help testing async block production.

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:52]: 👍

File: gradle.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-version=22.10.0-RC3-SNAPSHOT
+version=22.10.0-SNAPSHOT
```
Comment (macfarla) [inline gradle.properties:1]: are we doing this already

Comment (garyschulte) [reply gradle.properties]: fixed, thanks

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**